src/print/node-helpers.ts

```diff
@@ -0,0 +1,53 @@
+import { Node } from './nodes';
+import { inlineElements, TagName } from '../lib/elements';
+
+export function isInlineElement(node: Node) {
+  return node.type === 'Element' && inlineElements.includes(node.name as TagName);
+}
+
+export function isWhitespaceChar(ch: string) {
+  return ' \t\n\r'.indexOf(ch) >= 0;
+}
+
+export function canBreakAfter(node: Node) {
+  switch (node.type) {
+      case 'Text':
+          return isWhitespaceChar(node.raw[node.raw.length - 1]);
+      case 'Element':
+          return !isInlineElement(node);
+      default:
+          return true;
+  }
+}
+
+export function canBreakBefore(node: Node) {
+  switch (node.type) {
+      case 'Text':
+          return isWhitespaceChar(node.raw[0]);
+      case 'Element':
+          return !isInlineElement(node);
+      default:
+          return true;
+  }
+}
+
+export function isInlineNode(node: Node): boolean {
+  switch (node.type) {
+      case 'Text':
+          const text = node.raw || node.data;
+
+          return text === '' || text.trim() !== '';
```

Comment:
No, sorry, it's a confusing expression. An empty string should be treated as inline, because it doesn't trigger a line break. A string that just contains whitespace should not, because we can break there. Any other string is inline. 

I'll rewrite the expression to make it a bit clearer.

Reply:
Oh damn, I did not look close enough, it's `===` and `!==`, my bad 😄